packages/cli/src/config/extension.ts

```diff
@@ -253,6 +253,24 @@ export function loadExtension(
       )
       .filter((contextFilePath) => fs.existsSync(contextFilePath));
 
+    // IDs are created by hashing parts of the extension metadata in order to
+    // obfuscate any potentially sensitive information such as private
+    // git urls or project names.
+    const hash = createHash('sha256');
+    switch (installMetadata?.type) {
+      case 'git':
+      case 'github-release':
+        // For git and github-release installs, identify the extension based on the
+        // source it was installed from. This allows us to distinguish between
+        // extensions that happen to have the same name but are actually different.
+        hash.update(installMetadata.source);
+        break;
+      default:
+        // For all other install types, just use the name.
+        hash.update(config.name);
```

Comment:
Should we use the source here as well?

Reply:
Also I think when we log the extension ID for people, we can provide a warning if it is anything other than a git or github-release extension, that the ID is only as unique as the name.